### ISSUE TITLE
oneDNN finds wrong header for ArmPL

### ISSUE THREAD

**Issue by Zha0q1** [oneapi-src/oneDNN#1100]
I am working on a MXNet + oneDNN build on arm, with oneDNN integrated with ArmPL.
I have both ArmPL and OpenBlas installed in my environment and it seems that oneDNN would find the wrong header file for ArmPL.
```
[2021-06-17T20:00:16.022Z] -- Looking for sgemm_ - found
[2021-06-17T20:00:16.022Z] -- Found BLAS: /opt/arm/armpl_21.0_gcc-8.2/lib/libarmpl_lp64_mp.so;/usr/local/gcc-8.5.0/lib64/libgfortran.so;/opt/arm/armpl_21.0_gcc-8.2/lib/libamath.so;/usr/lib/aarch64-linux-gnu/libm.so  
[2021-06-17T20:00:16.022Z] -- Looking for cblas_sgemm
[2021-06-17T20:00:16.022Z] -- Looking for cblas_sgemm - found
[2021-06-17T20:00:16.022Z] -- Found CBLAS: /opt/arm/armpl_21.0_gcc-8.2/lib/libarmpl_lp64_mp.so;/usr/local/gcc-8.5.0/lib64/libgfortran.so;/opt/arm/armpl_21.0_gcc-8.2/lib/libamath.so;/usr/lib/aarch64-linux-gnu/libm.so
[2021-06-17T20:00:16.022Z] -- CBLAS include path: /work/mxnet/staticdeps/include
```
`CBLAS include path: /work/mxnet/staticdeps/include` this is the header path for OpenBLAS. So I tried to change https://github.com/oneapi-src/oneDNN/blob/83ebc40d86bc54f0f23e947235e53570eeacf254/cmake/blas.cmake#L75 from 
```
find_path(BLAS_INCLUDE_DIR ${CBLAS_HEADERS} $ENV{CPATH} ${BLAS_LIB_DIR}/../include ${BLAS_LIB_DIR}/../../include)
```
to
```
find_path(BLAS_INCLUDE_DIR ${CBLAS_HEADERS} HINTS $ENV{CPATH} ${BLAS_LIB_DIR}/../include ${BLAS_LIB_DIR}/../../include)
```
it seems to be able to find the correct header for me. Is this a CMake bug?

**Comment by Zha0q1** [comment]
Related MXNet PR is here https://github.com/apache/incubator-mxnet/pull/20364

**Comment by vpirogov** [comment]
+@nSircombe

**Comment by nSircombe** [comment]
Hi @Zha0q1,

Thanks for the report (and thanks @vpirogov for the heads-up). This isn't something we've encountered, but I usually have OpenBLAS in a module, which I suspect could hide this issue. I'll take a closer look.

**Comment by Zha0q1** [comment]
Hi @nSircombe I was able to workaround this by setting `set(BLAS_INCLUDE_DIR "/opt/arm/armpl_21.0_gcc-8.2/include_lp64_mp" CACHE STRING "Specifies BLAS include dir")` in MXNet. Yeah I think that would be a good solution, thanks! If this fix is merged I will not need to hardcode the include dir

**Comment by nSircombe** [comment]
Hi @Zha0q1 

I've raised a PR with the change (https://github.com/oneapi-src/oneDNN/pull/1111) if you'd like to give it a go.

**Comment by igorsafo** [comment]
#1111 was merged: 3b28c52
@Zha0q1  Could you please validate the changes on your side?

**Comment by Zha0q1** [comment]
Thanks for the fix! I have also worked around the issue
